main/src/ca/uwaterloo/flix/api/lsp/provider/completion/KeywordCompleter.scala

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2024 Alexander Dybdahl Troelsen
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.api.lsp.provider.completion
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.api.lsp.Index
+import ca.uwaterloo.flix.language.ast.TypedAst
+
+/**
+  * Completions for keywords
+  */
+object KeywordCompleter {
+  /**
+    * Miscellaneous keywords.
+    */
+  def getOtherKeywords(context: CompletionContext)(implicit flix: Flix, index: Index, root: TypedAst.Root): Iterable[Completion] =
+    List(
+      Completion.KeywordCompletion("with", Priority.highest("with")),
```

Comment:
Next turn Priority into an actual enum consisting of case objects.

Reply:
Does it make sense?

Reply:
Yep, it does. Currently on it :+1:

Reply:
if I wanna refactor priority entirely, then it might be better to do in a new PR

Reply:
or, what do you think?

Reply:
We can

Reply:
Do it in a separate PR